### PR TITLE
Add Layout component

### PR DESCRIPTION
**Pull request by Tijani-Dia**
This PR adds a layout component.

It's based on #7 because we need the SVG components here.

### CODE REVIEW

File: components/Layout/Footer/Footer.module.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    &__text {
+        @include font-size(xxs);
+        display: flex;
+        flex-direction: column;
+        position: relative;
+
+        @include media-query(medium) {
+            flex-direction: row;
+        }
```
Comment (jhancock532) [inline components/Layout/Footer/Footer.module.scss:31]: I like how you've organised the CSS rules here: include, then display, then flex-direction, then position, then media query 👍

File: components/Layout/Footer/Footer.module.scss [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        align-items: center;
+        width: auto;
+    }
+
+    .copyright {
+        margin-right: 0.25rem;
+
+        @include media-query(medium) {
+            font-weight: $weight--bold;
+        }
```
Comment (jhancock532) [inline components/Layout/Footer/Footer.module.scss:20]:
```suggestion
```
We can get rid of this extra style rule as `.copyright` is in a strong tag.

File: components/Layout/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { AreWeHeadlessYetLogo } from '../../SVG';
+import styles from './Footer.module.scss';
+import type FooterProps from './types';
+
+function getYear(date: string) {
+    const dateObject = new Date(date);
+    return dateObject.getFullYear();
+}
+
+export const Footer = ({ lastPublishedAt }: FooterProps) => (
```
Comment (jhancock532) [inline components/Layout/Footer/Footer.tsx:10]: love the camelCase update  🙌

File: components/Layout/Footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    <footer className={styles.footer}>
+        <div className={styles.footer__text}>
+            <div>
+                <strong className={styles.copyright}>
+                    © Are we headless yet?
+                </strong>
```
Comment (jhancock532) [inline components/Layout/Footer/Footer.tsx:16]: I see that this is supposed to render as not bold on mobile, but the `<strong>` tag adds font-weight bold by default - maybe replace this with a span?

File: components/Layout/Footer/Footer.module.scss (hunk 2 above)
Comment (jhancock532) [reply components/Layout/Footer/Footer.module.scss:20]: Correction: maybe we can keep this in, but replace the strong tag with a span instead (see below)

File: components/Layout/Footer/types.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+type FooterProps = {
+    lastPublishedAt: string;
+};
+
+export default FooterProps;
```
Comment (jhancock532) [inline components/Layout/Footer/types.ts:5]: Storing types separately is a good way of keeping file size down, do you feel it interferes with your development much? The projects I've worked on so far mostly keep the prop types beside the component, with the exception of the extended page types I've used in the careers site

File: components/Layout/FooterSlackCTA/FooterSlackCTA.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    <div className={styles.container}>
+        <div className={styles.emoji}>👂</div>
+        <div>
+            <div className={styles.cta__text}>Hear the latest news…</div>
+            <div className={styles.cta}>
+                Join{' '}
+                <a href="https://github.com/wagtail/wagtail/wiki/Slack">
+                    <strong className={styles.cta__tag}>#headless</strong>
+                </a>{' '}
+                on Slack
```
Comment (jhancock532) [inline components/Layout/FooterSlackCTA/FooterSlackCTA.tsx:13]: There's something up with the preview site Vercel deployed for this branch not loading the current code correctly - for this footer cta, the entire text is a link, not just the #headless.

image

File: components/Layout/HeadingSlackCTA/HeadingSlackCTA.module.scss [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        font-weight: $weight--extrabold;
+
+        @include media-query(large) {
+            flex-direction: column;
+        }
+
+        &__tag {
+            color: var(--color--accent);
+            font-weight: inherit;
+            margin: 0 0.1rem;
```
Comment (jhancock532) [inline components/Layout/HeadingSlackCTA/HeadingSlackCTA.module.scss:32]:
```suggestion
            margin: 0 0.2rem;
```
Just a tiny bit more margin here to give a proper space between the text

image

File: components/Layout/FooterSlackCTA/FooterSlackCTA.tsx (hunk 6 above)
Comment (jhancock532) [reply components/Layout/FooterSlackCTA/FooterSlackCTA.tsx:13]: The preview deployment for this branch has been taken down I think, Vercel is responding with a 404 when I try to view it. Maybe it defaults to an older deployment?

File: components/Layout/FooterSlackCTA/FooterSlackCTA.module.scss [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    &:before {
+        content: '';
+        position: absolute;
+        top: -1rem;
+        left: -2.5rem;
+        width: 75%;
+        border-color: var(--color--accent);
+        border-style: solid;
+        border-width: 4px;
```
Comment (jhancock532) [inline components/Layout/FooterSlackCTA/FooterSlackCTA.module.scss:49]:
```suggestion
        border: 4px solid var(--color--accent);
```

File: components/Layout/HeadingSlackCTA/HeadingSlackCTA.module.scss [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        line-height: 1.2;
+        font-weight: $weight--extrabold;
+
+        @include media-query(large) {
+            flex-direction: column;
+        }
+
+        &__tag {
+            color: var(--color--accent);
+            font-weight: inherit;
```
Comment (jhancock532) [inline components/Layout/HeadingSlackCTA/HeadingSlackCTA.module.scss:31]: As this is placed in a strong tag, is font-weight inherit needed here?

File: components/Layout/Layout.module.scss [hunk 10]
```diff
@@ -0,0 +1,13 @@
+.container {
+    padding: 0 2rem;
+}
```
Comment (jhancock532) [inline components/Layout/Layout.module.scss:3]: You might want to consider adding a max width and margin to this container 
```suggestion
.container {
    padding: 0 2rem;
    max-width: 1440px; // or $site-width;
    margin: 0 auto;
}
```

This means that the footer and footer cta don't get split up too far.

image

File: components/Layout/Layout.tsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+import styles from './Layout.module.scss';
+import type LayoutProps from './types';
+import HeadingSlackCTA from './HeadingSlackCTA';
+import FooterSlackCTA from './FooterSlackCTA';
+import Footer from './Footer';
+
+const Layout = ({ title, children, lastPublishedAt }: LayoutProps) => (
+    <>
+        <Head>
+            <title>{title}</title>
```
Comment (jhancock532) [inline components/Layout/Layout.tsx:11]: Nice positioning of `Head`, maybe add a ticket to codebase as a reminder to add social media metadata here later on.

**Review by jhancock532 — COMMENTED** [review]
Looking good, Vercel deployments don't seem to be previewing correctly but this might be because you've deleted or deactivated them.

File: components/Layout/Footer/Footer.tsx (hunk 4 above)
Comment (Tijani-Dia) [reply components/Layout/Footer/Footer.tsx:16]: See [this comment from Thibaud](https://github.com/wagtail/areweheadlessyet/pull/1#discussion_r797520016) suggesting that 
> it's a design inconsistency for this kind of text to have different styles across viewports.

So I think we want to render it bold on mobile as well.

File: components/Layout/Footer/types.ts (hunk 5 above)
Comment (Tijani-Dia) [reply components/Layout/Footer/types.ts:5]: > do you feel it interferes with your development much?

Not especially; some components come with a lot of typing so I preferred having the same approach (of having a separate file for types) in all components.

File: components/Layout/HeadingSlackCTA/HeadingSlackCTA.module.scss (hunk 7 above)
Comment (Tijani-Dia) [reply components/Layout/HeadingSlackCTA/HeadingSlackCTA.module.scss]: Added

File: components/Layout/HeadingSlackCTA/HeadingSlackCTA.module.scss (hunk 9 above)
Comment (Tijani-Dia) [reply components/Layout/HeadingSlackCTA/HeadingSlackCTA.module.scss:31]: The strong tag has `font-weight: bold;` whereas we want to have `font-weight: extrabold;` here.

File: components/Layout/Layout.tsx (hunk 11 above)
Comment (Tijani-Dia) [reply components/Layout/Layout.tsx:11]: I've added a ticket for adding social media metadata to Codebase.

File: components/Layout/Footer/Footer.module.scss (hunk 2 above)
Comment (Tijani-Dia) [reply components/Layout/Footer/Footer.module.scss:20]: See comment below

File: components/Layout/Layout.module.scss (hunk 10 above)
Comment (Tijani-Dia) [reply components/Layout/Layout.module.scss]: Added.

**Comment by Tijani-Dia** [comment]
This PR only adds the component definition without using them on the home page which is why the preview seems wrong.

**Review by patrickcuagan — COMMENTED** [review]
Thanks Tidiane - just one small comment!

**Pull request closed without merging**